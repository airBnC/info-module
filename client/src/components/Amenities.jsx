import PropTypes from 'prop-types';
import React from 'react';
import { SVGContainer } from './styles/Summary.styles';
import {
  Modal,
  ModalContent,
} from './styles/styles';
import {
  BasicSVG,
  DiningSVG,
  FacilitiesSVG,
  GuestSVG,
  BnbSVG,
  SafetySVG,
} from './SVG';
import {
  Wrapper,
  Title,
  CategoryHead,
  CategoryItemContainer,
  PreviewContainer,
  ShowAll,
  PreviewAmenity,
  PreviewText,
} from './styles/Amenities.styles';

const renderIcon = (category) => {
  if (category === 'Dining') {
    return (
      <SVGContainer>
        <DiningSVG />
      </SVGContainer>
    );
  } else if (category === 'Facilities') {
    return (
      <SVGContainer>
        <FacilitiesSVG />
      </SVGContainer>
    );
  } else if (category === 'Guest access') {
    return (
      <SVGContainer>
        <GuestSVG />
      </SVGContainer>
    );
  } else if (category === 'Bed and bath') {
    return (
      <SVGContainer>
        <BnbSVG />
      </SVGContainer>
    );
  } else if (category === 'Safety features') {
    return (
      <SVGContainer>
        <SafetySVG />
      </SVGContainer>
    );
  }
  return (
    <SVGContainer>
      <BasicSVG />
    </SVGContainer>
  );
};

const amenitiesCount = (amenities) => {
  let count = 0;
  for (let i = 0; i < amenities.length; i += 1) {
    count += amenities[i].category_items.length;
  }
  return count;
};

export default class Amenities extends React.Component {
  constructor(props) {
    super(props);
    this.handleAmenitiesDisplay = this.handleAmenitiesDisplay.bind(this);
    this.handleAmenitiesHide = this.handleAmenitiesHide.bind(this);
    this.state = {
      displayModal: 'none',
    };
  }

  handleAmenitiesDisplay() {
    this.setState({
      displayModal: 'flex',
    });
  }

  handleAmenitiesHide() {
    this.setState({
      displayModal: 'none',
    });
  }

  render() {
    return (
      <Wrapper>
        <PreviewContainer>
          <PreviewText>
            Amenitites
          </PreviewText>
          {
            this.props.amenities.items.map(item => (
              <PreviewAmenity
                key={item.category_head}
              >
                {renderIcon(item.category_head)}
                {item.category_items[0].amenity_description}
              </PreviewAmenity>
            ))
          }
          <ShowAll
            onClick={this.handleAmenitiesDisplay}
            className="modal-display"
          >
            Show all {amenitiesCount(this.props.amenities.items)} amenities
          </ShowAll>
        </PreviewContainer>
        <Modal
          displayModal={this.state.displayModal}
          onClick={this.handleAmenitiesHide}
          className="modal-hide"
        >
          <ModalContent>
            <Title>
              Amenities
            </Title>
            <Wrapper>
              {
                this.props.amenities.items.map(item => (
                  <Wrapper
                    key={item.category_head}
                  >
                    <CategoryHead>
                      {item.category_head}
                    </CategoryHead>
                    {
                      item.category_items.map(categoryItem => (
                        <CategoryItemContainer
                          key={categoryItem.amenity_description}
                        >
                          <PreviewText>
                            {categoryItem.amenity_description}
                          </PreviewText>
                          <PreviewText>
                            {categoryItem.supplemental_description}
                          </PreviewText>
                        </CategoryItemContainer>
                      ))
                    }
                  </Wrapper>
                ))
              }
            </Wrapper>
          </ModalContent>
        </Modal>
      </Wrapper>);
  }
}

Amenities.propTypes = {
  amenities: PropTypes.shape({
    id: PropTypes.number,
    items: PropTypes.arrayOf(PropTypes.shape({
      category_head: PropTypes.string,
      category_items: PropTypes.arrayOf(PropTypes.shape({
        amenity_description: PropTypes.string,
        supplemental_description: PropTypes.string,
      })),
    })),
  }),
};

Amenities.defaultProps = {
  amenities: {
    id: 1,
    items: [
      {
        category_head: 'String',
        category_items: [
          {
            amenity_description: 'String',
            supplemental_description: 'String',
          },
        ],
      },
    ],
  },
};
