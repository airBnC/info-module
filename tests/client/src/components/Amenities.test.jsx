import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Amenities from '../../../../client/src/components/Amenities.jsx';

Enzyme.configure({ adapter: new Adapter() });

const amenities = {
  id: 1,
  items: [
    {
      category_head: 'Mock Category Head',
      category_items: [
        {
          amenity_description: 'Mock Category Item',
          supplemental_description: 'Mock Supplemental Description',
        },
      ],
    },
  ],
};

describe('Description', () => {
  it('should render correctly', () => {
    const output = Enzyme.shallow(<Amenities
      amenities={amenities}
    />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
