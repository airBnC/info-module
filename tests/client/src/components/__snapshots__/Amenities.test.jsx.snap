// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render correctly 1`] = `
<div
  id="amenities"
  style={
    Object {
      "border": "solid black 1px",
    }
  }
>
  <h1>
    Amenities
  </h1>
  <div
    key="Mock Category Head"
  >
    <h3>
      Mock Category Head
    </h3>
    <div
      key="Mock Category Item"
    >
      <h4>
        Mock Category Item
      </h4>
      <p>
        Mock Supplemental Description
      </p>
    </div>
  </div>
</div>
`;
